tests/unit/test_serialization.py

```diff
@@ -45,6 +48,121 @@ async def test_happy_flow(default_config_path, mock_registry_client):
         assert 1 in o.ids_to_schemas
 
 
+def test_avro_json_flatten_unions():
```

Comment:
Maybe split into multiple tests - one for `record`, one for `list` etc. Now there are two tests for `array` and a test for `map` between those.